### DISCUSSION TITLE
How to show commits of the current file?

### DISCUSSION THREAD

**Discussion by kkoomen** [NeogitOrg/neogit#1141] [General]
Hi,

As far as I see, I can only see the complete commit history, but I want to have some kind of shortcut inside a buffer to see the commits with diffs for that buffer. Is this possible?

Thanks.

**Comment by CKolkey** [comment]
I'd use https://github.com/sindrets/diffview.nvim with `:DiffviewFileHistory %` for a view of a file-specific history with diffs.

In Neogit, you could filter the log buffer by filename.

**Reply by kkoomen** [reply]
@CKolkey how would I do filter in neogit by buffer filename?

**Reply by CKolkey** [reply]
Same as on git cli - open the log popup via `l` then hit `--` and provide one or more filepaths 

Then open the log view you want via `l` or `h` or `e`

**Reply by kkoomen** [reply]
I've created a new folder with git: `a.txt` has one commit, `b.txt` is in two commits.

Using FZF for `a.txt`, it shows only one, as it should:

<img width="1364" alt="image" src="https://github.com/NeogitOrg/neogit/assets/10693490/bbc8b9b3-faeb-4501-a1ab-f3de68d44631">

Using the mentioned keystroked you mentioned, I get this:

<img width="548" alt="image" src="https://github.com/NeogitOrg/neogit/assets/10693490/56d64150-a5ec-4046-b6ed-2408a9138d1b">

Which is not right. To make sure I've done it right, here's to show that it is indeed filtered on a.txt:

<img width="528" alt="image" src="https://github.com/NeogitOrg/neogit/assets/10693490/191e88b8-1cba-477d-b405-57a769147056">

Any ideas?

**Reply by CKolkey** [reply]
So, here's how I tried to reproduce this - correct me if I'm mistaken.

```bash
cd /tmp
mkdir test
cd test
git init
touch a
touch b
git add a
git commit -m "add a"
git add b
git commit -m "add b"
echo "foo" > a
git add a
git commit -m "update a"
```

Within neogit, I hit `l`, then `--` and select "a". Then `l` again. That lists two commits - "add a" then "update a".

Close the buffer with `q`, `l` again, `--` and select "b". `l` again, and it lists one commit "add b". Both are what I would expect.

**Reply by kkoomen** [reply]
Seems to work now, I don't know why it didn't work before. Thanks!
